pkg/cloud/aws/actuators/machine/utils.go

```diff
@@ -204,10 +204,14 @@ func EncodeProviderStatus(codec codec, awsStatus *providerconfigv1.AWSMachinePro
 
 // IsMaster returns true if the machine is part of a cluster's control plane
 func IsMaster(machine *clusterv1.Machine) bool {
+	var isMaster bool
+	if machineRole, exists := machine.ObjectMeta.Labels[providerconfigv1.MachineRoleLabel]; exists && machineRole == "master" {
+		isMaster = true
+	}
 	if machineType, exists := machine.ObjectMeta.Labels[providerconfigv1.MachineTypeLabel]; exists && machineType == "master" {
-		return true
+		isMaster = true
```

Comment:
Why can't the function return `true` directly?

Reply:
Thanks, way better. But as you pointed out the type check can go anyhow.